pkg/kubectl/cmd/util/factory.go

```diff
@@ -123,26 +140,38 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		Client: func() (*client.Client, error) {
 			return clients.ClientForVersion("")
 		},
+		ExperimentalClient: func() (*client.ExperimentalClient, error) {
+			return experimentalClient, experimentalClientErr
+		},
 		ClientConfig: func() (*client.Config, error) {
 			return clients.ClientConfigForVersion("")
 		},
 		RESTClient: func(mapping *meta.RESTMapping) (resource.RESTClient, error) {
-			client, err := clients.ClientForVersion(mapping.APIVersion)
-			if err != nil {
-				return nil, err
+			switch groupForResource(mapping.Resource) {
+			case apiGroup:
+				client, err := clients.ClientForVersion(mapping.APIVersion)
+				if err != nil {
+					return nil, err
+				}
+				return client.RESTClient, nil
+			case experimentalGroup:
+				client, err := experimentalClient, experimentalClientErr
```

Comment:
See my other comment above - you can't fail in Factory initialization by reading the config.